### DISCUSSION TITLE
404 on localhost:8080 with the example `docker run` command, modified to work with localhost

### DISCUSSION THREAD

**Discussion by gremo** [shlinkio/discussions#131] [Help wanted] [answered]
Following https://shlink.io/documentation/install-docker-image/ and slighly modified to make it work on localhost for testing purposes:

```bash
docker run \
    --name my_shlink \
    -p 8080:8080 \
    -e DEFAULT_DOMAIN=localhost \
    -e IS_HTTPS_ENABLED=false \
    shlinkio/shlink:stable
```

Opening localhost:8080 results ina 404 error. The Docker log:

>  2023-11-22T12:57:41+0000        INFO    server          [2023-11-22T12:57:41.163245+00:00] [5fc100d4-d382-4ab2-a090-96be6b49ce41] Access.INFO - GET / 404 979

**Comment by acelaya** [comment, marked as answer]
Yes, that's expected. Shlink always returns a 404 in the base URL unless the `base_url` [special redirect](https://shlink.io/documentation/advanced/special-redirects/) is configured.

Also, in your case it should be `DEFAULT_DOMAIN=localhost:8080`, but that's unrelated to the 404.

**Reply by gremo** [reply]
Oh I see. In that case, how do I access the shortner itself?

**Reply by acelaya** [reply]
With "the shortener" I guess you mean a UI?

Shlink is [architected](https://shlink.io/documentation/advanced/shlink-architecture/) as a headless API, so the main service does not provide a UI.

The options you have are:

* Use the CLI: https://shlink.io/documentation/command-line-interface/entry-point/
* Use the shlink-web-client instance hosted in https://app.shlink.io
  shlink-web-client runs 100% in the browser, so you can use that instance as much as you want https://shlink.io/documentation/shlink-web-client/

**Reply by gremo** [reply]
Thank you @acelaya for your help!
